service/test/BatchServerTest.cpp

```diff
@@ -160,39 +179,77 @@ TEST_F(BatchServerTest, get_server_pid)
     EXPECT_EQ(m_server_pid, server_pid);
 }
 
+/**
+ * @test Check BatchServerImp::get_server_key() in isolation
+ */
 TEST_F(BatchServerTest, get_server_key)
 {
     std::string key = m_batch_server->server_key();
     EXPECT_EQ(std::to_string(m_client_pid), key);
 }
 
+/**
+ * @test Check that BatchServerImp::stop_batch() sends a SIGTERM signal to the server.
+ */
 TEST_F(BatchServerTest, stop_batch)
 {
     EXPECT_TRUE(m_batch_server->is_active());
     EXPECT_CALL(*m_posix_signal,
-                sig_queue(m_server_pid,
-                          SIGTERM,
-                          BatchStatus::M_MESSAGE_TERMINATE));
+                sig_queue(m_server_pid, SIGTERM, BatchStatus::M_MESSAGE_TERMINATE));
     m_batch_server->stop_batch();
     EXPECT_FALSE(m_batch_server->is_active());
 }
 
+/**
+ * @test Check that BatchServerImp::stop_batch() throws an exception if it is called from the child process.
+ *       Then check the catch block in BatchServerImp::stop_batch() by forcing a failure of POSIXSignalImp::sig_queue().
+ */
+TEST_F(BatchServerTest, stop_batch_exception)
+{
+    GEOPM_EXPECT_THROW_MESSAGE(m_batch_server_child_process->stop_batch(),
+                               GEOPM_ERROR_RUNTIME,
+                               "BatchServerImp::stop_batch(): must be called from parent process, not child process");
+
+    EXPECT_CALL(*m_posix_signal,
+                sig_queue(m_server_pid, SIGTERM, BatchStatus::M_MESSAGE_TERMINATE))
+        .Times(1)
+        .WillRepeatedly(
+            /// @throws geopm::Exception to simulate a failure of the sigqueue(2) system call in POSIXSignalImp::sig_queue()
+            [](){
+                throw geopm::Exception("POSIXSignal(): POSIX signal function call sigqueue() returned an error",
+                           EINVAL, __FILE__, __LINE__);
+            }
+        );
```

Comment:
```suggestion
        .WillRepeatedly(Throw(geopm::Exception("POSIXSignal(): POSIX signal function call sigqueue() returned an error",
                              EINVAL, __FILE__, __LINE__)));
```

Please remove all these lambdas and use the ```Throw()``` action.
https://google.github.io/googletest/reference/actions.html